### ISSUE TITLE
False positive for RSpec/PendingWithoutReason with block syntax

### ISSUE THREAD

**Issue by tomchipchase** [rubocop/rubocop-rspec#1562]
We've got one last false positive with this new cop:

When the skip has a block, it gets incorrectly flagged:
```
it "works" do
  # do some stuff

  skip 'not yet implemented' do
    # do some checks
  end
end
```

This works fine though (its not quite the same thing if there was some other stuff after the skip block, but fine for our usecase).
```
it "works" do
  # do some stuff

  skip 'not yet implemented'
  # do some checks
end
```

**Comment by pirj** [comment]
Passing a block to `skip`/`pending` seems to be a misuse, as [they don't accept blocks](https://github.com/rspec/rspec-core/blob/79ec0fd5b5ebd13031a10960f293b8a4a43cb734/lib/rspec/core/pending.rb#L91) ([docs](https://relishapp.com/rspec/rspec-core/v/3-9/docs/pending-and-skipped-examples/skip-examples)).

> its not quite the same thing if there was some other stuff after the skip block

Can you give an example when it's not the same thing?

`skip` stops example execution immediately. The presence of code after the `skip` statement doesn't matter.
`pending` doesn't stop, but its purpose is to make sure that the example fails, so it would be weird to skip anything in a block that never gets called.

I'd prefer to keep flagging this case, even though it's not this cop's responsibility, but rather an idea for some other cop.

**Comment by tomchipchase** [comment]
Fair enough! We were misusing this feature then.

**Comment by pirj** [comment]
Provided there is a cop that is watching over in-example ‘pending’/‘skip’ with blocks, I’m fine with fixing this false positive in ‘PendingWiyhoutReason’ that additionally avoids duplicate offences, and to completely split responsibilities.

**Comment by pirj** [comment]
Fixed in #1566
